db/active_replicator_common_collections.go

```diff
@@ -160,3 +171,14 @@ func (arc *activeReplicatorCommon) forEachCollection(callback func(*activeReplic
 	}
 	return nil
 }
+
+// getFilteredChannels returns the filtered channels.
+// collectionIdx can be nil if replicating without collections enabled.
+func (config ActiveReplicatorConfig) getFilteredChannels(collectionIdx *int) []string {
+	if collectionIdx != nil {
+		if len(config.CollectionsChannelFilter)-1 >= *collectionIdx {
+			return config.CollectionsChannelFilter[*collectionIdx]
+		}
+	}
+	return config.FilterChannels
+}
```

Comment:
I think this function works but it's almost by accident: I think an easier way of writing this is:

```
        // default collection (should we validate config.CollectionsEnabled when we call this?)
	if collectionIdx == nil {
             return config.FilterChannels
        }
	if len(config.CollectionsChannelFilter)-1 >= *collectionIdx {
			return config.CollectionsChannelFilter[*collectionIdx]
		}
        // filter not present for collections, show all channels
	return nil
}
```

The reason it works before is that `config.FilterChannels` was prevalidated to have empty list.

Reply:
Actually, I want it to behave like this so that `config.FilterChannels` can act across all collections (option 2 in earlier comment) as the fallback (assuming the earlier validation ensured that these two fields are mutually exclusive).